project/src/main/webapp/event-details.js

```diff
@@ -1,32 +1,47 @@
-const eventHost = 'test@example.com'; // hard-coded event host value
-let loginState = null;
-let currentUser = null;
-
-window.onload = function onLoad() {
+window.onload = async function onLoad() {
   getEventDetails();
-  checkLoginStatus();
-  getVolunteeringOpportunities();
-  populateOpportunitiesDropdown();
+  const eventHost = await getEventHost();
+  const loginStatus = await getLoginStatus();
+  populateVolunteeringOpportunitiesUI(eventHost, loginStatus);
+  showCreateOpportunityLink(eventHost, loginStatus);
+  setSignupAction();
 };
 
 
 /**
- * Check the current login status to update
- * isLoggedIn and currentUser.
+ * Returns the current login status.
  */
-async function checkLoginStatus() {
+async function getLoginStatus() {
   const response = await fetch('/login-status');
   const loginStatus = await response.json();
-  loginState = loginStatus.loginState;
-  currentUser = loginStatus.userEmail;
+  return loginStatus;
 }
 
 /**
- * Adds the volunteering opportunities to the event card.
+ * Adds the volunteering opportunities to the event card and the dropdown
+ * in the signup form.
+ * @param {string} eventHost email of the host of the event
+ * @param {Object} loginStatus status of
+ *      current user
  */
-async function getVolunteeringOpportunities() {
-  const response = await fetch('/event-volunteering-data');
+async function populateVolunteeringOpportunitiesUI(eventHost, loginStatus) {
+  const eventId = getEventId();
+
+  const response = await fetch(`/event-volunteering-data?event-id=${eventId}`);
   const opportunities = await response.json();
+  showVolunteeringOpportunities(opportunities, eventHost, loginStatus);
+  populateOpportunitiesDropdown(opportunities);
+}
+
+/**
+ * Should the volunteering opportunities of the event card and if the
+ * eventHost is the current user, add the edit link for each opportunity.
+ * @param {Object} opportunities
+ * @param {string} eventHost email of the host of the event
+ * @param {Object} loginStatus status of current user
+ */
+async function showVolunteeringOpportunities(
+    opportunities, eventHost, loginStatus) {
   for (const key in opportunities) {
```

Comment:
Use for-of loops when you can 
https://google.github.io/styleguide/jsguide.html#features-for-loops

Reply:
Done.